src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+    private static final int OPERATION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final int DEFAULT_AMOUNT = 0;
+    private static final int REQUIRED_PARTS_COUNT = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            Map<String, Integer> operationTotals = readData(fromFileName);
+            int supplyTotal = operationTotals.getOrDefault(SUPPLY, DEFAULT_AMOUNT);
+            int buyTotal = operationTotals.getOrDefault(BUY, DEFAULT_AMOUNT);
+            int result = supplyTotal - buyTotal;
+            writeReport(toFileName, supplyTotal, buyTotal, result);
+        } catch (Exception e) {
+            throw new RuntimeException("Error processing files " + fromFileName + toFileName, e);
+        }
+    }
+
+    private Map<String, Integer> readData(String fileName) {
+        Map<String, Integer> operationTotals = new HashMap<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(COMMA);
+                if (parts.length == REQUIRED_PARTS_COUNT) {
+                    String operation = parts[OPERATION_INDEX].trim();
+                    try {
+                        int amount = Integer.parseInt(parts[AMOUNT_INDEX].trim());
+                        operationTotals.put(operation, operationTotals
+                                .getOrDefault(operation, DEFAULT_AMOUNT) + amount);
+                    } catch (Exception e) {
+                        throw new RuntimeException("Error parsing amount in line: " + line, e);
+                    }
```

Comment:
Why did you choose to use try-catch here?

Reply:
Because there can be a NumberFormatException. For example if amount in our csv file will be "a123". Moreover, after amount there can be a space or Byte Order Mark and this exception will occur)

Reply:
There is trim() but it only removes characters with codes from 0 to 32 so there is a possibilitty for getting an exception here

Reply:
> Why did you choose to use try-catch here?

Because there can be a NumberFormatException. For example if amount in our csv file will be "a123". Moreover, after amount there can be other char like BOM) There is trim() but it only removes characters with codes from 0 to 32 so there is a possibilitty for getting an exception here

Reply:
> Why did you choose to use try-catch here?

Myb i should use replaceAll against trim() to not create a try catch block